command/agent/agent.go

```diff
@@ -75,16 +78,20 @@ type clientServer interface {
 // mode, it runs a full Consul server. In client-only mode, it only forwards
 // requests to other Consul servers.
 type Agent struct {
+	// id is an optional log prefix.
+	id string
```

Comment:
I think it makes more sense to just provide a properly configured logger which is used everywhere. Since I've fixed the logger passing we can move this into the `TestAgent`